### PR TITLE
hubble:Add unit test for Pop with closed stop chan

### PR DESCRIPTION
**Pull request by Jianlin-lv**
Signed-off-by: Jianlin Lv <Jianlin.Lv@arm.com>

Related: #12178

### CODE REVIEW

**Comment by gandro** [comment]
Awesome, thanks! I assume this requires https://github.com/cilium/cilium/pull/12257 to be merged first?

**Comment by Jianlin-lv** [comment]
> Awesome, thanks! I assume this requires to be rebased on #12257 first?
Yes, I should submit this PR based on #12257

**Comment by gandro** [comment]
Travis fails as expected :tada: Once #12257 is merged, this needs a rebase on master.

**Comment by Jianlin-lv** [comment]
> Travis fails as expected 🎉 Now that #12257 is merged, this needs a rebase on master.

okay

**Review by gandro — APPROVED** [review]

**Comment by gandro** [comment]
test-me-please

**Review by michi-covalent — APPROVED** [review]

File: pkg/hubble/peer/buffer_test.go [hunk 1]
```diff
@@ -82,3 +82,20 @@ func TestBufferPop(t *testing.T) {
...
+	buf := newBuffer(max)
+	assert.NotNil(t, buf)
+	assert.Equal(t, 0, buf.Len())
+	assert.Equal(t, 0, buf.Cap())
+
+	for i := 0; i < max; i++ {
+		assert.NoError(t, buf.Push(&peerpb.ChangeNotification{}))
+		assert.Equal(t, i+1, buf.Len())
+	}
+	close(buf.stop)
```
Comment (rolinh) [inline pkg/hubble/peer/buffer_test.go:97]:
```suggestion
	buf.Close()
```
Please, use `buf.Close()` to close the `stop` chan.

**Review by rolinh — CHANGES_REQUESTED** [review]

File: pkg/hubble/peer/buffer_test.go (hunk 1 above)
Comment (Jianlin-lv) [reply pkg/hubble/peer/buffer_test.go:97]: buf.Close() cleans up b.buf, this causes b.stop chan to be performed in the first select statement in Pop();
It is not the purpose of this unit case.

Comment (rolinh) [reply pkg/hubble/peer/buffer_test.go:97]: Mmh, I see with `buf.Close()` you only have a 50/50 chance to trigger the real problem whereas by artificially closing the channel, you ensure that there is a check for a closed `stop` chan.

**Review by rolinh — APPROVED** [review]

**Pull request merged**